tests/functional/docker-compose.yml

```diff
@@ -0,0 +1,78 @@
+# -*- coding: utf-8 -*-
+#
+# This file is part of hepcrawl.
+# Copyright (C) 2015, 2016, 2017 CERN.
+#
+# hepcrawl is a free software; you can redistribute it and/or modify it
+# under the terms of the Revised BSD License; see LICENSE file for
+# more details.
+
+version: '2'
+
+services:
+  test_wsp:
+    image: hepcrawl_docker
+    environment: &env_variables
+      - APP_BROKER_URL=amqp://guest:guest@rabbitmq:5672//
+      - APP_CELERY_RESULT_BACKEND=amqp://guest:guest@rabbitmq:5672//
+      - APP_CRAWLER_HOST_URL=http://scrapyd:6800
+      - APP_API_PIPELINE_TASK_ENDPOINT_DEFAULT=tests.functional.tasks.submit_results
+    command: py.test tests/functional/WSP/test_wsp.py
+    volumes: &static_volume
+      - /home/${USER}/.virtualenvs/hepcrawl:/hepcrawl_venv/
+      - ${PWD}:/code/
+    links:
+      - "rabbitmq"
+      - "celery"
+      - "scrapyd"
+    depends_on:
+      - rabbitmq
+      - scrapyd
+      - celery
+      - ftp_server
+      - scrapyd_deploy
+
+  celery:
+    image: hepcrawl_docker
+    environment: *env_variables
+    command: celery worker --events --app tests.functional.tasks --loglevel=debug
+    volumes: *static_volume
+    links:
+      - "rabbitmq"
+      - "ftp_server"
+
+  scrapyd:
+    image: hepcrawl_docker
+    environment: *env_variables
+    command: bash -c "rm -f twistd.pid && exec scrapyd"
+    volumes: *static_volume
+    ports:
+      - "6800:6800"
```

Comment:
None of the ports entries are really needed here, as they are only needed when opening ports from the current host (your laptop) to the containers, not between containers.